cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -174,8 +175,14 @@ func createPacketManagerRest(configReader io.Reader, discoverOpts cloudprovider.
 		cfg.Global.ClusterName = opts.ClusterName
 	}
 
+	packetAuthToken := os.Getenv("PACKET_AUTH_TOKEN")
+	if len(packetAuthToken) == 0 {
+		klog.Fatalf("PACKET_AUTH_TOKEN is required and missing")
```

Comment:
I wonder what's the general recommendation here. Should process crash (and restart) if configuration is wrong? Or should it run (as it's capable of, it's not segfaulting or anyting) and just log the error and block?

Reply:
The only way to heal this is to add the environment variable to the pod, which would trigger a restart. It seems reasonable to me to fail early and hard (so long as this failure doesn't disturb the rest of the autoscaler functionality 🤔 ... Does it?)

Reply:
Yeah, I went with this approach to help make it more obvious that the required configuration is missing. The other cloud providers do similar as well.

Reply:
Going back to @displague's question around disturbing the rest of the autoscaler functionality, since the autoscaler can only be configured for a single provider at a given time it would technically disturb the rest of the autoscaler functionality, but that functionality would already be broken in that it wouldn't be able to perform any of the operations required to function anyway with the lack of credentials to interact with the provider API